### PR TITLE
Log 1651: Restricting Elasticsearch cluster privileges

### PR DESCRIPTION
**Pull request by Red-GV**
### Description
This PR adds a a restrictive SCC similar to the Fluentd implementation (https://github.com/openshift/cluster-logging-operator/pull/1003), with a few allowances to not completely break everything. The changes are:

- A few volume mounts being changed to read only
- An scc which drops all capabilities and forbids all systcl
- Containers are not allow to escalate privileges 

/cc @periklis 
/assign @igor-karpukhin 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-1651

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -15,19 +15,6 @@ LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3
...
-LABEL \
-    com.redhat.component="elasticsearch-operator" \
-    version="v1.1" \
-    name="elasticsearch-operator" \
-    License="Apache-2.0" \
-    io.k8s.display-name="elasticsearch-operator bundle" \
-    io.k8s.description="bundle for the elasticsearch-operator" \
-    summary="This is the bundle for the elasticsearch-operator" \
-    maintainer="AOS Logging <aos-logging@redhat.com>"
-
```
Comment (periklis) [inline bundle.Dockerfile:30]: Please revert this change. This is an unfortunate hindsight of `make bundle` still.

File: internal/manifests/securitycontextconstraints/securitycontextconstraints.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func ConstraintsEqual(current, desired *securityv1.SecurityContextConstraints) bool {
+	return equality.Semantic.DeepEqual(current.Volumes, desired.Volumes) &&
+		equality.Semantic.DeepEqual(current.ReadOnlyRootFilesystem, desired.ReadOnlyRootFilesystem)
+}
+
+func MutateConstraints(current, desired *securityv1.SecurityContextConstraints) {
+	current.Volumes = desired.Volumes
+	current.ReadOnlyRootFilesystem = desired.ReadOnlyRootFilesystem
+}
```
Comment (periklis) [inline internal/manifests/securitycontextconstraints/securitycontextconstraints.go:83]: Can you elaborate a bit on why we check/mutate only these two fields? If there is some sort of immutability/third-party-ownership on the other SCC fields, we should rename `ContraintsEqual` to something `ConstrainVolumeEqual`. Take a look on `manifests/services` for a good example.

File: internal/elasticsearch/securitycontextconstraints.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	builder.WithAllowPrivilegeEscalation(false)
+	builder.WithDefaultAllowPrivilegeEscalation(false)
+	builder.WithVolumes([]securityv1.FSType{"configMap", "secret", "emptyDir", "persistentVolumeClaim"})
+	builder.WithRequiredDropCapabilities([]corev1.Capability{"CHOWN", "DAC_OVERRIDE", "FSETID", "FOWNER", "SETGID", "SETUID", "SETPCAP", "NET_BIND_SERVICE", "KILL"})
+	builder.WithRunAsUserOptions(securityv1.RunAsUserStrategyOptions{
+		Type: securityv1.RunAsUserStrategyRunAsAny,
+	})
+	builder.WithSELinuxContextOptions(securityv1.SELinuxContextStrategyOptions{
+		Type: securityv1.SELinuxStrategyRunAsAny,
+	})
```
Comment (periklis) [inline internal/elasticsearch/securitycontextconstraints.go:28]: Security settings are very hard to read and maintain over time. I suggest you add per line a two-liner comment to explain why we set this as is and what's the impact on the ES container.

File: internal/utils/utils.go [hunk 4]
```diff
@@ -167,6 +167,11 @@ func GetInt64(value int64) *int64 {
 	return &i
 }
 
+func GetBool(value bool) *bool {
+	b := value
+	return &b
+}
+
```
Comment (periklis) [inline internal/utils/utils.go:174]: I suggest to stop adding custom helper as this one and start leveraging `"k8s.io/utils/pointer"` that supports `pointer.BoolPtr(true)` and others.

**Review by periklis — CHANGES_REQUESTED** [review]
Awesome work! Keep :rocket: !

A general hint:
-  Usage of new apis (e.g. `securityv1`) requires registration into the default client scheme as here: https://github.com/openshift/elasticsearch-operator/blob/master/main.go#L48

File: internal/manifests/securitycontextconstraints/securitycontextconstraints.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	current.AllowPrivilegedContainer = desired.AllowPrivilegedContainer
+	current.AllowHostDirVolumePlugin = desired.AllowHostDirVolumePlugin
+	current.ReadOnlyRootFilesystem = desired.ReadOnlyRootFilesystem
+	current.Volumes = desired.Volumes
+	current.ForbiddenSysctls = desired.ForbiddenSysctls
+	current.RequiredDropCapabilities = desired.RequiredDropCapabilities
+	current.AllowPrivilegeEscalation = desired.AllowPrivilegeEscalation
+	current.DefaultAllowPrivilegeEscalation = desired.DefaultAllowPrivilegeEscalation
+	current.RunAsUser = desired.RunAsUser
+	current.SELinuxContext = desired.SELinuxContext
```
Comment (periklis) [inline internal/manifests/securitycontextconstraints/securitycontextconstraints.go:83]: Is this list complete?

**Review by periklis — COMMENTED** [review]
Looks really good! Awesome job on inline docs. If this one required any changes on openshift/origin-aggregated-logging, we should discuss how/when to backport till 5.1

File: internal/manifests/securitycontextconstraints/securitycontextconstraints.go (hunk 5 above)
Comment (Red-GV) [reply internal/manifests/securitycontextconstraints/securitycontextconstraints.go]: It should now be the exhaustive list of changes as described in the security api.

**Comment by Red-GV** [comment]
Thanks, @periklis! No OAL changes were needed for this PR.
